visualizer/src/topology.js

```diff
@@ -197,17 +80,163 @@ module.exports = class Topology {
         return d;
     }
 
-    moduleConnectionNumber(mName) {
-        const connectionsNumber = {};
-        for (const moduleName of Object.keys(this.data.modules)) {
-            connectionsNumber[moduleName] = 0;
+    createConnectionTypeMap() {
+        const connectionTypeMap = {};
+        for (const source of Object.keys(this.data.modules)) {
+            connectionTypeMap[source] = {};
+            for (const target of Object.keys(this.data.modules)) {
+                connectionTypeMap[source][target] = 5;
+            }
+        }
+        return connectionTypeMap;
+    }
+
+    configureConnections() {
+        const connectionTypeMap = this.createConnectionTypeMap();
+        for (const [portName, portInfo] of Object.entries(this.data.portmap)) {
+            for (const [moduleName, ports] of Object.entries(this.data.modules)) {
+                if (ports.write_ports !== '' && portName in ports.write_ports) {
+                    this.configureConnection(portName, moduleName, portInfo, connectionTypeMap);
+                }
+            }
+        }
+    }
+}
+
+class jsPlumbConfig extends BaseConfig {
+    configureModules(modulemap, parent) {
+        for (const [moduleName, children] of Object.entries(modulemap)) {
+            let module = {}
+            module = this.newModule(moduleName);
+            if (children !== "") {
+                this.initParent(module);
+                this.configureModules(children, moduleName);
+            }
+            if (parent !== "") {
+                this.instance.addToGroup(`${parent}_group`, module);
+            }
+        }

```

Comment:
Similarly, you can define "configureModule" function to keep "configureModules" generalized.